localization/util/pose_initializer/src/pose_initializer_core.cpp

```diff
@@ -88,25 +99,22 @@ bool PoseInitializer::serviceInitial(
   add_height_pose_msg_ptr->pose.covariance[4 * 6 + 4] = 0.01;
   add_height_pose_msg_ptr->pose.covariance[5 * 6 + 5] = 1.0;
 
-  geometry_msgs::PoseWithCovarianceStamped::Ptr aligned_pose_msg_ptr(
-    new geometry_msgs::PoseWithCovarianceStamped);
-  const bool succeeded_align = callAlignService(*add_height_pose_msg_ptr, aligned_pose_msg_ptr);
+  auto aligned_pose_msg_ptr = std::make_shared<geometry_msgs::msg::PoseWithCovarianceStamped>();
+  const bool succeeded_align = callAlignService(add_height_pose_msg_ptr, aligned_pose_msg_ptr);
 
   if (succeeded_align) {
-    initial_pose_pub_.publish(*aligned_pose_msg_ptr);
-    return true;
+    initial_pose_pub_->publish(*aligned_pose_msg_ptr);
   } else {
-    return false;
+    throw std::runtime_error("Aligning poses did not succeed.");
```

Comment:
Service handlers don't return a bool anymore. It is recommended by the migration guide to throw an exception instead.